File: imagemover/app/src/main/java/com/atmaca/imagemover/MainActivity.java
package com.atmaca.imagemover;

import android.Manifest;
import android.app.Activity;
import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.graphics.Typeface;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.os.Environment;
import android.provider.Settings;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

public final class MainActivity extends Activity {
    private static final int STORAGE_REQUEST = 1907;

    private TextView statusView;
    private Button permissionButton;
    private boolean settingsOpened;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setGravity(Gravity.CENTER);
        root.setPadding(dp(24), dp(24), dp(24), dp(24));

        TextView title = new TextView(this);
        title.setText("ATMACA 1907");
        title.setTextSize(30f);
        title.setTypeface(Typeface.DEFAULT_BOLD);
        title.setGravity(Gravity.CENTER);
        root.addView(title, new LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.WRAP_CONTENT
        ));

        statusView = new TextView(this);
        statusView.setTextSize(17f);
        statusView.setGravity(Gravity.CENTER);
        LinearLayout.LayoutParams statusParams = new LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.WRAP_CONTENT
        );
        statusParams.topMargin = dp(22);
        root.addView(statusView, statusParams);

        permissionButton = new Button(this);
        permissionButton.setText("TÜM DOSYALARA ERİŞİMİ AÇ");
        permissionButton.setOnClickListener(v -> openStorageAccessSettings());
        LinearLayout.LayoutParams buttonParams = new LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.MATCH_PARENT,
                LinearLayout.LayoutParams.WRAP_CONTENT
        );
        buttonParams.topMargin = dp(22);
        root.addView(permissionButton, buttonParams);

        setContentView(root);
    }

    @Override
    protected void onResume() {
        super.onResume();
        ensureAccessAndStart();
    }

    private void ensureAccessAndStart() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
            if (Environment.isExternalStorageManager()) {
                showRunningAndStart();
            } else {
                statusView.setText("İlk çalıştırmada Android'in ‘Tüm dosyalara erişim’ iznini bir kez aç.\nİzin verince taşıma otomatik başlayacak.");
                permissionButton.setVisibility(View.VISIBLE);
                if (!settingsOpened) {
                    settingsOpened = true;
                    openStorageAccessSettings();
                }
            }
            return;
        }

        if (checkSelfPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED) {
            showRunningAndStart();
        } else {
            statusView.setText("Depolama izni gerekiyor. İzin verince taşıma otomatik başlayacak.");
            permissionButton.setVisibility(View.GONE);
            requestPermissions(new String[]{
                    Manifest.permission.READ_EXTERNAL_STORAGE,
                    Manifest.permission.WRITE_EXTERNAL_STORAGE
            }, STORAGE_REQUEST);
        }
    }

    private void openStorageAccessSettings() {
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R) {
            requestPermissions(new String[]{
                    Manifest.permission.READ_EXTERNAL_STORAGE,
                    Manifest.permission.WRITE_EXTERNAL_STORAGE
            }, STORAGE_REQUEST);
            return;
        }

        Intent appIntent = new Intent(
                Settings.ACTION_MANAGE_APP_ALL_FILES_ACCESS_PERMISSION,
                Uri.parse("package:" + getPackageName())
        );
        try {
            startActivity(appIntent);
        } catch (ActivityNotFoundException unavailable) {
            startActivity(new Intent(Settings.ACTION_MANAGE_ALL_FILES_ACCESS_PERMISSION));
        }
    }

    private void showRunningAndStart() {
        permissionButton.setVisibility(View.GONE);
        statusView.setText("Görseller bulunur bulunmaz Pictures/1907 klasörüne taşınıyor.\nAynı isimli dosyalar tek dosyada birleşir.");
        Intent serviceIntent = new Intent(this, MoveService.class);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            startForegroundService(serviceIntent);
        } else {
            startService(serviceIntent);
        }
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        if (requestCode == STORAGE_REQUEST) {
            ensureAccessAndStart();
        }
    }

    private int dp(int value) {
        return Math.round(value * getResources().getDisplayMetrics().density);
    }
}

File: filemanager/app/src/main/kotlin/com/atmaca/files/MainActivity.kt
package com.atmaca.files

import android.Manifest
import android.app.Activity
import android.content.Intent
import android.content.pm.PackageManager
import android.graphics.Color
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.view.Gravity
import android.widget.Button
import android.widget.GridLayout
import android.widget.LinearLayout
import android.widget.TextView

class MainActivity : Activity() {
    private val prefs by lazy { getSharedPreferences("atmaca_files", MODE_PRIVATE) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestMediaPermissions()
        buildUi()
    }

    private fun buildUi() {
        val navy = Color.rgb(7, 25, 58)
        val yellow = Color.rgb(255, 220, 0)
        val root = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setPadding(dp(18), dp(22), dp(18), dp(18))
            setBackgroundColor(navy)
        }
        root.addView(TextView(this).apply {
            text = "ATMACA DOSYALAR"
            textSize = 27f
            setTextColor(yellow)
            setTypeface(null, 1)
        })
        root.addView(TextView(this).apply {
            text = "Hafif dosya yöneticisi"
            textSize = 14f
            setTextColor(Color.LTGRAY)
            setPadding(0, dp(4), 0, dp(18))
        })

        val grid = GridLayout(this).apply { columnCount = 2; rowCount = 3 }
        val cats = listOf(
            "🖼 Görseller" to "images", "🎬 Videolar" to "videos",
            "🎵 Ses" to "audio", "📄 Belgeler" to "documents",
            "📦 APK" to "apks", "⬇ İndirilenler" to "downloads"
        )
        cats.forEach { (label, key) ->
            val b = Button(this).apply {
                text = label; textSize = 15f; setTextColor(Color.WHITE)
                setBackgroundColor(Color.rgb(18, 48, 92))
                setOnClickListener { startActivity(Intent(this@MainActivity, CategoryActivity::class.java).putExtra("category", key)) }
            }
            grid.addView(b, GridLayout.LayoutParams().apply {
                width = 0; height = dp(72); columnSpec = GridLayout.spec(GridLayout.UNDEFINED, 1f)
                setMargins(dp(5), dp(5), dp(5), dp(5))
            })
        }
        root.addView(grid)

        root.addView(TextView(this).apply {
            text = "Depolama"
            textSize = 19f; setTextColor(Color.WHITE); setTypeface(null, 1)
            setPadding(0, dp(18), 0, dp(8))
        })
        root.addView(actionButton("DAHİLİ / SD KART EKLE", yellow, navy) { pickStorage() })
        val saved = prefs.getString("root_uri", null)
        if (saved != null) {
            root.addView(actionButton("SEÇİLİ DEPOLAMAYI AÇ", Color.WHITE, navy) {
                startActivity(Intent(this, BrowserActivity::class.java).putExtra("uri", saved).putExtra("isRoot", true))
            })
        }
        root.addView(TextView(this).apply {
            text = "Tam disk taraması yapmaz; yalnız açtığın klasörü listeler."
            textSize = 12f; setTextColor(Color.LTGRAY); gravity = Gravity.CENTER_HORIZONTAL
            setPadding(0, dp(18), 0, 0)
        })
        setContentView(root)
    }

    private fun pickStorage() {
        startActivityForResult(Intent(Intent.ACTION_OPEN_DOCUMENT_TREE).addFlags(
            Intent.FLAG_GRANT_READ_URI_PERMISSION or Intent.FLAG_GRANT_WRITE_URI_PERMISSION or Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION
        ), 41)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == 41 && resultCode == RESULT_OK) {
            val uri = data?.data ?: return
            val flags = data.flags and (Intent.FLAG_GRANT_READ_URI_PERMISSION or Intent.FLAG_GRANT_WRITE_URI_PERMISSION)
            runCatching { contentResolver.takePersistableUriPermission(uri, flags) }
            prefs.edit().putString("root_uri", uri.toString()).apply()
            startActivity(Intent(this, BrowserActivity::class.java).putExtra("uri", uri.toString()).putExtra("isRoot", true))
            buildUi()
        }
    }

    private fun requestMediaPermissions() {
        if (Build.VERSION.SDK_INT >= 33) {
            val perms = arrayOf(Manifest.permission.READ_MEDIA_IMAGES, Manifest.permission.READ_MEDIA_VIDEO, Manifest.permission.READ_MEDIA_AUDIO)
            if (perms.any { checkSelfPermission(it) != PackageManager.PERMISSION_GRANTED }) requestPermissions(perms, 90)
        } else if (checkSelfPermission(Manifest.permission.READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
            requestPermissions(arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE), 90)
        }
    }

    private fun actionButton(label: String, bg: Int, fg: Int, click: () -> Unit) = Button(this).apply {
        text = label; textSize = 15f; setTextColor(fg); setTypeface(null, 1); setBackgroundColor(bg)
        setOnClickListener { click() }
    }
    private fun dp(v: Int) = Math.round(v * resources.displayMetrics.density)
}
